homeassistant/components/openai_conversation/entity.py

```diff
@@ -143,70 +144,119 @@ def _format_tool(
 
 
 def _convert_content_to_param(
-    content: conversation.Content,
+    chat_content: Iterable[conversation.Content],
 ) -> ResponseInputParam:
     """Convert any native chat message for this agent to the native format."""
     messages: ResponseInputParam = []
-    if isinstance(content, conversation.ToolResultContent):
-        return [
-            FunctionCallOutput(
-                type="function_call_output",
-                call_id=content.tool_call_id,
-                output=json.dumps(content.tool_result),
+    reasoning_summary: list[str] = []
+
+    for content in chat_content:
+        if isinstance(content, conversation.ToolResultContent):
+            messages.append(
+                FunctionCallOutput(
+                    type="function_call_output",
+                    call_id=content.tool_call_id,
+                    output=json.dumps(content.tool_result),
+                )
             )
-        ]
-
-    if content.content:
-        role: Literal["user", "assistant", "system", "developer"] = content.role
-        if role == "system":
-            role = "developer"
-        messages.append(
-            EasyInputMessageParam(type="message", role=role, content=content.content)
-        )
-
-    if isinstance(content, conversation.AssistantContent) and content.tool_calls:
-        messages.extend(
-            ResponseFunctionToolCallParam(
-                type="function_call",
-                name=tool_call.tool_name,
-                arguments=json.dumps(tool_call.tool_args),
-                call_id=tool_call.id,
+            continue
+
+        if content.content:
+            role: Literal["user", "assistant", "system", "developer"] = content.role
+            if role == "system":
+                role = "developer"
+            messages.append(
+                EasyInputMessageParam(
+                    type="message", role=role, content=content.content
+                )
             )
-            for tool_call in content.tool_calls
-        )
+
+        if isinstance(content, conversation.AssistantContent):
+            if content.tool_calls:
+                messages.extend(
+                    ResponseFunctionToolCallParam(
+                        type="function_call",
+                        name=tool_call.tool_name,
+                        arguments=json.dumps(tool_call.tool_args),
+                        call_id=tool_call.id,
+                    )
+                    for tool_call in content.tool_calls
+                )
+
+            if content.thinking_content:
+                reasoning_summary.append(content.thinking_content)
+
+            if isinstance(content.native, dict):
```

Comment:
Guard this so it's only included when the `content.agent_id == self.entity_id`

Reply:
Yes, good point. I am actually going to guard by `DOMAIN`, as I did for the google one: https://github.com/home-assistant/core/pull/150347/files#diff-108da9672b1da500cca9a35a05a9690bb10bdef1eb36696cc5fb53c60f34b202R233

Reply:
Fixed with a native object